### PR TITLE
Autocomplete: deselect previously selected item using backspace

### LINKED ISSUE #2776: Autocomplete: Allow user to deselect previously selected item using backspace
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/trimble-oss/modus-web-components/blob/main/CONTRIBUTING.md)

### Describe the issue

**Summary**
Provide option to remove the selected item in autocomplete component.

**Steps to reproduce**
- Enable multi select in [autocomplete component](https://modus-web-components.trimble.com/?path=/story/user-inputs-autocomplete--multiple-selection)
- Select one or more option from the list
- Press Backspace button in keyboard

**Expected**
- Should remove the last selected item

**Actual**
- Item is not removed and user have to use the mouse to remove the option. 

### Reduced test cases

_No response_

### What operating system(s) are you seeing the problem on?

_No response_

### What browser(s) are you seeing the problem on?

_No response_

### What is the issue regarding ?

@trimble-oss/modus-web-components

### What version of npm package are you using ?

0.35.0

### Priority

Medium

### What product/project are you using Modus Components for ?

modus-internal

### What is your team/division name ?

modus-internal

### Are you willing to contribute ?

None

### Are you using Modus Web Components in production ?

_No response_

### PR DESCRIPTION
**Pull request by prashanthr6383**
## Description

References 
Fixes #2776 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.tsx [hunk 1]
```diff
@@ -304,13 +304,28 @@ export class ModusAutocomplete {
...
+    if (e.key !== 'Backspace' || !this.multiple) {
+      return;
+    }
+
+    if (!this.getValueAsString() && this.selectedChips.length > 0) {
+      e.stopPropagation();
+      e.preventDefault();
+
+      this.selectedChips = this.selectedChips.slice(0, -1);
+      this.valueChange.emit(this.selectedChips.map((opt) => opt.value));
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.tsx:317]: No need to emit valueChange as it is supposed to be emitted only when the text input changes.

Comment (prashanthr6383) [reply stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.tsx:310]: @prashanth-offcl As we discussed, The valueChange would be a breaking change so therefore skipping this change and rest of them are done.

File: stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.tsx [hunk 2]
```diff
@@ -304,13 +304,21 @@ export class ModusAutocomplete {
     this.focusOptionItem();
   };
 
+  handleBackspaceKeyDown = (e) => {
```
Comment (prashanth-offcl) [inline stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.tsx:307]: @prashanthr6383 can you keep the name as handleInputKeyDown and check for defaultPrevented

Comment (prashanthr6383) [reply stencil-workspace/src/components/modus-autocomplete/modus-autocomplete.tsx]: @prashanth-offcl done.

**Review by prashanth-offcl — APPROVED** [review]

**Pull request merged**